Sources/SpeziConsent/Views/OnboardingConsentView.swift

```diff
@@ -0,0 +1,271 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+#if os(macOS)
+import AppKit
+#endif
+import Foundation
+import PDFKit
+import Spezi
+import SpeziOnboarding
+import SpeziViews
+import SwiftUI
+
+
+/// Onboarding view to display markdown-based consent documents that can be signed and exported.
+///
+/// The ``OnboardingConsentView`` provides a convenient onboarding `View` for the display of markdown-based documents that can be
+/// signed using a family and given name and a hand drawn signature.
+///
+/// Furthermore, the `View` includes an export functionality, enabling users to share and store the signed consent form.
+/// The exported consent form PDF is received via the `action` closure on the ``OnboardingConsentView/init(markdown:action:title:currentDateInSignature:exportConfiguration:)``.
+///
+/// The ``OnboardingConsentView`` builds on top of the SpeziOnboarding ``ConsentDocument``
+/// by providing a more developer-friendly, convenient API with additional functionalities like the share consent option.
+///
+/// ```swift
+/// OnboardingConsentView(
+///     markdown: {
+///         Data("This is a *markdown* **example**".utf8)
+///     },
+///     action: { exportedConsentPdf in
+///         // The action that should be performed once the user has provided their consent.
+///         // Closure receives the exported consent PDF to persist or upload it.
+///     },
+///     title: "Consent",   // Configure the title of the consent view
+///     exportConfiguration: .init(paperSize: .usLetter),   // Configure the properties of the exported consent form
+///     currentDateInSignature: true   // Indicates if the consent signature should include the current date
+/// )
+/// ```
+public struct OnboardingConsentView: View {
```

Comment:
Wondering if we should name this to `ConsentView`; especially as we move this outside of the Onboarding package?

Reply:
it's intentionally still called `OnboardingConsentView`, since it internally wraps its contents in an `OnboardingView` and is intended to be used in an OnboardingStack.

We have the `ConsentDocumentView` for scenarios where you just want to display a `ConsentDocument`, without having any Onboarding-specific UI/logic involved.

Reply:
Sounds fair. 

Regarding the API differences: This will be a breaking change anyways, so more than happy to align it and change elements to be as good as possible.